modules/gradio_ui.py

```diff
@@ -2,61 +2,73 @@
 import requests
 import json
 
-conversation_id=None
+class gradioUI:
+    def __init__(self,
+                 ols_url="http://127.0.0.1:8080/ols",
+                 mount_path="/ui",
+                 conversation_id=None) -> None:
+        # class variable
+        self.ols_url=ols_url
+        self.mount_path=mount_path
+        self.conversation_id = conversation_id
+    
+        # ui specific
+        use_history=gr.Checkbox(value=True, label="Use history")
+        self.ui = gr.ChatInterface(self.yaml_gen,
+                                    additional_inputs=[
+                                        use_history
+                                    ])
 
-def yaml_gen(prompt,history,use_history):
-    global conversation_id
-    # URL of the HTTP endpoint
-    url = "http://127.0.0.1:8080/ols"
+    def yaml_gen(self, prompt, history, use_history=None):
+        # URL of the HTTP endpoint
+        url = self.ols_url
```

Comment:
what's the point of making a copy of this value?

Reply:
moving the function into a class without modifying the function

Reply:
is there a reason not to modify the function though?  let's not carry cruft in that we don't need